server/s3_log.cc

```diff
@@ -21,6 +21,18 @@
 #include "s3_log.h"
 #include "s3_option.h"
 
+enum {
+  KB = 1 << 10,
+};
+
+static thread_local char log_buffer[10*KB];
+char *__log_buff() {
+  return log_buffer;
+}
+size_t __log_buff_sz() {
+  return 10*KB - 1;
```

Comment:
This code is a dirty prototype. Don't think about it as a full fledged implementation. For sure it can be related by sizeof() and it'll not be slower than 10*KB-1 as it's a compile time constant.